### PR TITLE
[Docs] Translate data transform docs.

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

Add data transform doc translation.

## Modification

Add data transform doc translation. And modify the intersphinx link to mmcv.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/conf.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
     'python': ('https://docs.python.org/3', None),
     'numpy': ('https://numpy.org/doc/stable', None),
     'torch': ('https://pytorch.org/docs/stable/', None),
-    'mmcv': ('https://mmcv.readthedocs.io/en/dev-2.x/', None),
+    'mmcv': ('https://mmcv.readthedocs.io/en/2.x/', None),
```
Comment (mzr1996) [inline docs/en/conf.py:61]: The dev-2.x doc is unavailable.

File: docs/en/advanced_tutorials/data_transform.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+In config files, we can compose multiple data transforms as a list, called a data pipeline. And the data
+pipeline is an argument of the dataset.
+
+Usually, a data pipeline consists of the following parts:
+
+1. Data loading, use [`LoadImageFromFile`](mmcv.transforms.LoadImageFromFile) to load image files.
+2. Label loading, use [`LoadAnnotations`](mmcv.transforms.LoadAnnotations) to load the bboxes, semantic segmentation and keypoint annotations.
+3. Data processing and augmentation, like [`RandomResize`](mmcv.transforms.RandomResize).
+4. Data formatting, we use different data transforms for different tasks. And the data transform for specified
+   task is implemented in the correspondding repository. For example, the data formatting transform for image
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/data_transform.md:43]:
```suggestion
   task is implemented in the corresponding repository. For example, the data formatting transform for image
```

File: docs/en/advanced_tutorials/data_transform.md [hunk 3]
```diff
@@ -1,3 +1,156 @@
...
+pre-processing & augmentation and data formatting.
+
+### Data Loading
+
+To support loading large-scale dataset, usually we won't load all dense data during dataset construction, but
+only load the file path of these data. Therefore, we need to load these data in the data pipeline.
+
+|                     Data Transforms                      |                                     Functionality                                     |
+| :------------------------------------------------------: | :-----------------------------------------------------------------------------------: |
+| [`LoadImageFromFile`](mmcv.transforms.LoadImageFromFile) |                          Load images according to the path.                           |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/data_transform.md:84]: These links failed in [readthedocs build](https://mmengine--737.org.readthedocs.build/en/737/advanced_tutorials/data_transform.html)

File: docs/zh_cn/advanced_tutorials/data_transform.md [hunk 4]
```diff
@@ -65,34 +65,34 @@ test_dataloader = dict(
 
 为了支持大规模数据集的加载，通常在数据集初始化时不加载数据，只加载相应的路径。因此需要在数据流水线中进行具体数据的加载。
 
-|                               数据变换类                                |                   功能                    |
-| :---------------------------------------------------------------------: | :---------------------------------------: |
-| [`LoadImageFromFile`](https://mmcv.readthedocs.io/en/2.x/api.html#TODO) |             根据路径加载图像              |
-|  [`LoadAnnotations`](https://mmcv.readthedocs.io/en/2.x/api.html#TODO)  | 加载和组织标注信息，如 bbox、语义分割图等 |
+|                        数据变换类                        |                   功能                    |
+| :------------------------------------------------------: | :---------------------------------------: |
+| [`LoadImageFromFile`](mmcv.transforms.LoadImageFromFile) |             根据路径加载图像              |
```
Comment (C1rN09) [inline docs/zh_cn/advanced_tutorials/data_transform.md:70]: 这里的链接也是

File: docs/en/migration/transform.md [hunk 5]
```diff
@@ -1,3 +1,162 @@
...
+    <th>MMCV (new)</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td><code>LoadImageFromFile</code></td>
+    <td>Join the 'img_prefix' and 'img_info.filename' field to find the path of images and loading.</td>
+    <td>Join the 'img_prefix' and 'img_info.filename' field to find the path of images and loading. Support
+    specifying the order of channels.</td>
+    <td>Load images from 'img_path'. Support ignoring failed loading and specifying decode backend.</td>
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply docs/en/migration/transform.md:36]: Why do we need to change the single quotation marks to double?

File: docs/zh_cn/conf.py [hunk 6]
```diff
@@ -50,7 +50,6 @@
     'sphinx.ext.napoleon',
     'sphinx.ext.viewcode',
     'sphinx.ext.autosectionlabel',
-    'sphinx_markdown_tables',
```
Comment (mzr1996) [inline docs/zh_cn/conf.py:53]: `sphinx_markdown_tables` is useless and conflicts with `myst`.
It also breaks the references in tables.

**Review by C1rN09 — APPROVED** [review]

File: docs/en/migration/transform.md [hunk 7]
```diff
@@ -1,3 +1,162 @@
...
+
+In OpenMMLab 2.0, to make the data transform classes more extensible, we use `transform` method instead of
+`__call__` method to implement data transformation, and all data transform classes should inherit the
+[`mmcv.transforms.BaseTransfrom`](mmcv.transforms.BaseTransfrom) class. And you can still use these data
+transform classes by calling.
+
+A tutorial to implement a data transform class can be found in the [Data Transform](../advanced_tutorials/data_element.md).
+
+In addition, we move some common data transform classes from every repositories to MMCV, and in this document,
+we will compare the functionalities, usages and implementations between the original data transform classes (in [MMClassification v0.23.2](https://github.com/open-mmlab/mmclassification/tree/v0.23.2), [MMDetection v2.25.1](https://github.com/open-mmlab/mmdetection/tree/v2.25.1)) and the new data transform classes (in [MMCV v2.0.0rc1](https://github.com/open-mmlab/mmcv/tree/dev-2.x))
```
Comment (zhouzaida) [inline docs/en/migration/transform.md:17]:
```suggestion
we will compare the functionalities, usages and implementations between the original data transform classes (in [MMClassification v0.23.2](https://github.com/open-mmlab/mmclassification/tree/v0.23.2), [MMDetection v2.25.1](https://github.com/open-mmlab/mmdetection/tree/v2.25.1)) and the new data transform classes (in [MMCV v2.0.0rc1](https://github.com/open-mmlab/mmcv/tree/2.x))
```

**Comment by zhouzaida** [comment]
This PR can be merged after resolving comments and conflicts.

File: docs/en/tutorials/data_transform.md [hunk 8]
```diff
@@ -1,3 +1,156 @@
...
+4. Data formatting, we use different data transforms for different tasks. And the data transform for specified
+   task is implemented in the corresponding repository. For example, the data formatting transform for image
+   classification task is `PackClsInputs` and it's in MMClassification.
+
+Here, taking the classification task as an example, we show a typical data pipeline in the figure below. For
+each sample, the basic information stored in the dataset is a dictionary as shown on the far left side of the
+figure, after which, every blue block represents a data transform, and in every data transform, we add some new fields (marked in green) or update some existing fields (marked in orange) in the data dictionary.
+
+<div align=center>
+<img src="https://user-images.githubusercontent.com/26739999/187157681-ac4dcac8-3543-4bfe-ab30-9aa9e56d4900.jpg" width="90%"/>
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply docs/en/tutorials/data_transform.md]: I don't know how to remove it

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**